### PR TITLE
feat: [CI-4663]:  compare Yaml executions, view compiled Yaml - in build and deployment

### PR DESCRIPTION
**Pull request by udayvunnam**
##### Summary:

- compare Yaml executions in build and deployment and pipeline executions
- view compiled Yaml for executions in build and deployment
- context for comparing executions since they need to be available at different tree levels like header, card, checkbox, and execution actions
- Backend API integration
- Integration and Unit tests added for coverage

##### Jira Links:

https://harness.atlassian.net/browse/CI-4663
https://harness.atlassian.net/browse/PIE-3878

##### Screenshots:
Compare Yaml's
Kapture 2022-06-19 at 18 11 57

View Compiled Yaml:
Kapture 2022-06-19 at 18 18 39

### CODE REVIEW

File: src/modules/10-common/components/InputDatePicker/__test__/__snapshots__/InputDatePicker.test.tsx.snap [hunk 1]
```diff
@@ -35,7 +35,7 @@ exports[`Launch Button test Input date picker render  1`] = `
               >
                 <input
                   class="bp3-input"
-                  placeholder="1/15/1970 - 2/15/1970"
+                  placeholder="15/01/1970 - 15/02/1970"
```
Comment (udayvunnam) [inline src/modules/10-common/components/InputDatePicker/__test__/__snapshots__/InputDatePicker.test.tsx.snap:38]: Haven't changed anything related to this test, this snapshot was out of sync

File: src/modules/70-pipeline/pages/pipeline-deployment-list/__tests__/execution-list.json [hunk 2]
```diff
@@ -42,6 +42,88 @@
         "runningStagesCount": 0,
         "failedStagesCount": 1,
         "totalStagesCount": 0
+      },
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/pipeline-deployment-list/__tests__/execution-list.json:45]: added two more executions two add test cases for compare executions

File: src/modules/75-cd/components/ServiceDetails/__tests__/PipelineExecutions.test.tsx [hunk 3]
```diff
@@ -47,40 +48,37 @@ const noData = {
   }
 }
 
+const renderPipelineExecutions = (): RenderResult =>
+  render(
+    <TestWrapper>
+      <ExecutionCompareProvider>
```
Comment (udayvunnam) [inline src/modules/75-cd/components/ServiceDetails/__tests__/PipelineExecutions.test.tsx:54]: extract to common render function wrapped in ExecutionCompareProvider, unit tests

File: src/modules/70-pipeline/pages/pipeline-deployment-list/__tests__/PipelineDeploymentList.test.tsx [hunk 4]
```diff
@@ -432,4 +443,40 @@ describe('Test Pipeline Deployment list', () => {
       }
     })
   })
+
+  test('should compare executions of various deployments', async () => {
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/pipeline-deployment-list/__tests__/PipelineDeploymentList.test.tsx:447]: Integration test at PipelineDeploymentList for compare executions

**Comment by udayvunnam** [comment]
dockerize

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/70-pipeline/components/ExecutionCompiledYaml/ExecutionCompiledYaml.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            <PageError message={(error.data as Error)?.message || error.message} onClick={refetch as any} />
+          ) : (
+            <MonacoEditor
+              data-testid="execution-compiled-yaml-viewer"
+              width="100%"
+              height="100%"
+              value={data?.data?.executionYaml}
+              language={'yaml'}
+              options={
+                {
```
Comment (YounixIO) [inline src/modules/70-pipeline/components/ExecutionCompiledYaml/ExecutionCompiledYaml.tsx:85]: Move config to a constant

File: src/modules/70-pipeline/components/ExecutionCompareYamls/ExecutionCompareYamls.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+          </Layout.Horizontal>
+          <MonacoDiffEditor
+            data-testid="execution-compare-yaml-viewer"
+            width="100%"
+            height="100%"
+            original={dataOne?.data?.executionYaml}
+            value={dataTwo?.data?.executionYaml}
+            language={'yaml'}
+            options={
+              {
```
Comment (YounixIO) [inline src/modules/70-pipeline/components/ExecutionCompareYamls/ExecutionCompareYamls.tsx:97]: Move to a constant

**Review by YounixIO — APPROVED** [review]

File: src/services/pipeline-ng/index.tsx [hunk 7]
```diff
@@ -9193,7 +9260,7 @@ export const createPipelineV2Promise = (
     ResponsePipelineSaveResponse,
     Failure | Error,
     CreatePipelineV2QueryParams,
-    CreateOverlayInputSetForPipelineBodyRequestBody,
+    UpdateInputSetForPipelineBodyRequestBody,
```
Comment (ishantmehtaharness) [inline src/services/pipeline-ng/index.tsx:9263]: Is this change expected?

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

File: src/services/pipeline-ng/index.tsx (hunk 7 above)
Comment (udayvunnam) [reply src/services/pipeline-ng/index.tsx:9263]: Thanks for the catch, I took the latest swagger provided by Backend to generate these types. Is there any automated flow that validates the swagger used in the frontend(to generate queries) with current backend `develop` changes?

cc: @kapilkumawat86

**Review by vardanbansal-harness — APPROVED** [review]

**Pull request merged**